samples/msgext-search-sso-config/nodejs/bots/teamsMessagingExtensionsSearchAuthConfigBot.js

```diff
@@ -48,28 +47,30 @@ class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
 
     // Overloaded function. Receives invoke activities with Activity name of 'composeExtension/queryLink'
     async handleTeamsAppBasedLinkQuery(context, query) {
-        const magicCode =
-            query.state && Number.isInteger(Number(query.state))
-                ? query.state
-                : '';
-        const tokenResponse = await context.adapter.getUserToken(
-            context,
-            this.connectionName,
-            magicCode
-        );
+        const userTokenClient = context.turnState.get(context.adapter.UserTokenClientKey);
+            const magicCode =
+                context.state && Number.isInteger(Number(context.state))
+                    ? context.state
+                    : '';
+
+            const tokenResponse = await userTokenClient.getUserToken(
+                context.activity.from.id,
+                this.connectionName,
+                context.activity.channelId,
+                magicCode
+            );
 
         if (!tokenResponse || !tokenResponse.token) {
             // There is no token, so the user has not signed in yet.
-
             // Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions
-            const signInLink = await context.adapter.getSignInLink(
-                context,
-                this.connectionName
+            const { signInLink } = await userTokenClient.getSignInResource(
+                this.connectionName,
+                context.activity
             );
 
             return {
                 composeExtension: {
-                    type: 'auth',
+                    type: 'silentAuth',
```

Comment:
if you change here, once user first use link unfurling without consent before, there will show no login prompt.

Reply:
I have tested its working..
FYI - 
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/2b70ee9d-4fd5-43f0-8cc0-307caf8a2ca0)
**Click login -** 
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/27a591a5-a1e4-4e64-a650-159072270da3)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/f5b5be1c-0026-4a9d-8211-797b7d234ec5)

Reply:
great! 
but can you share the request and respone in F12? 
once there recieved the 412 response  but show nothing anymore.
I tried your branch code but show no any login prompt. is there any other settings on teams?
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/fc95680e-445f-4ba9-afad-51c2752317fc)

Reply:
**Please check this document and follow all steps. - [Setup for Bot SSO](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-conversation-sso-quickstart/BotSSOSetup.md) and check your bot-oAuth connection, botID, and password.**

Reply:
yes, I follow all the steps, and query in message extension can show the prompt login link.
so i think the link unfurling has something different with query in message extension
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/036f2d56-8492-4367-aad8-cb3bb99552ad)

Reply:
once change the `silentAuth` to `auth` in `handleTeamsAppBasedLinkQuery`, it works fine
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/0b408bb8-34d5-4fdb-8d0e-40f07ea589a2)

Reply:
Its working **silentAuth and auth**, we are maintaining contestant perforce.

Reply:
Actually, link unfurling not working with `silentAuth`, please try again and show your F12 as same as following picture
![Uploading image.png…]()